### PR TITLE
Link clicking

### PR DESCRIPTION
**Pull request by ncalexan**
@jsantell, this should address #344.

### CODE REVIEW

File: test/webdriver/test-link-clicking.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
+// Any copyright is dedicated to the Public Domain.
+// http://creativecommons.org/publicdomain/zero/1.0/
+
+/* eslint prefer-arrow-callback: 0 */
+
+import expect from 'expect';
+import { quickTest } from '../../build-config';
+import Driver from '../utils/driver';
+import * as selectors from '../../app/ui/browser/selectors/index';
```
Comment (jsantell) [inline test/webdriver/test-link-clicking.js]: `import { getPages, getCurrentPageIndex }` as they're the only ones used

File: test/webdriver/test-link-clicking.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+const STARTING_TAB_COUNT = 1;
```
Comment (jsantell) [inline test/webdriver/test-link-clicking.js]: In other tests, I regret having this as it makes the assertions harder to follow -- I think we can assume one tab on opening for these tests

File: test/utils/driver.js [hunk 3]
```diff
@@ -83,11 +87,18 @@ const Driver = {
...
-      JSON.stringify(require('electron').ipcRenderer.store.getState()))
-      .then(res => JSON.parse(res.value));
+    const oldHandle = await this.setTargetToBrowserWindow();
+
+    try {
+      return await this.client.execute(() =>
+        JSON.stringify(require('electron').ipcRenderer.store.getState()))
+        .then(res => JSON.parse(res.value));
+    } finally {
+      // Here's a situation where a decorator would be useful.
```
Comment (jsantell) [inline test/utils/driver.js]: Agreed -- after running a few tests, the overhead of switching contexts isn't much at all. We're slowly piecing together the story for these tests so should be good to abstract this out shortly.

File: app/ui/browser/views/page/page.jsx [hunk 4]
```diff
@@ -86,6 +90,13 @@ Page.propTypes = {
 export default Page;
 
 function addListenersToWebView(webview, pageAccessor, dispatch) {
+  webview.addEventListener('new-window', (e) => {
+    // TODO: differentiate more thoroughly based on e.disposition.
+    console.log(`disposition ${e.disposition}`);
```
Comment (jsantell) [inline app/ui/browser/views/page/page.jsx]: remove console.log

File: app/ui/browser/views/page/page.jsx [hunk 5]
```diff
@@ -86,6 +90,13 @@ Page.propTypes = {
 export default Page;
 
 function addListenersToWebView(webview, pageAccessor, dispatch) {
+  webview.addEventListener('new-window', (e) => {
```
Comment (jsantell) [inline app/ui/browser/views/page/page.jsx]: Should add a comment here about what this event is and the disposition property

File: test/webdriver/test-link-clicking.js [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+    await Driver.blur();
+    await Driver.navigate(`${Driver.fixturesURL}/links.html`);
+    await Driver.waitForCurrentTabLoaded();
+    expect(true).toBe(true, 'Navigated to links.html');
+
+    await Driver.setTargetByURL(`${Driver.fixturesURL}/links.html`);
+    await Driver.client.click('#this-tab');
+
+    const text = await Driver.client.getText('#hello');
```
Comment (jsantell) [inline test/webdriver/test-link-clicking.js]: Might need to wait until this is visible, not sure if getText handles that

File: app/ui/browser/views/page/page.jsx [hunk 7]
```diff
@@ -27,6 +27,11 @@ const PAGE_STYLE = Style.registerStyle({
   // Mark this as the relative anchor for floating children (e.g. search bar).
   position: 'relative',
   flex: 1,
+  '&:not(.active-browser-page)': {
```
Comment (jsantell) [inline app/ui/browser/views/page/page.jsx]: If this is from `https://github.com/electron/electron/blob/master/docs/api/web-view-tag.md#css-styling-notes` we should comment it as such

**Comment by jsantell** [comment]
r+

File: test/webdriver/test-link-clicking.js (hunk 2 above)
Comment (ncalexan) [inline test/webdriver/test-link-clicking.js]: Thanks!  I've inlined it throughout.

File: app/ui/browser/views/page/page.jsx (hunk 4 above)
Comment (ncalexan) [inline app/ui/browser/views/page/page.jsx]: Done.

File: app/ui/browser/views/page/page.jsx (hunk 5 above)
Comment (ncalexan) [inline app/ui/browser/views/page/page.jsx]: I've added a link to the documentation.

File: app/ui/browser/views/page/page.jsx (hunk 7 above)
Comment (ncalexan) [inline app/ui/browser/views/page/page.jsx]: Good point.  Done.

**Comment by ncalexan** [comment]
Landed.

**Pull request closed without merging**